packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol

```diff
@@ -54,16 +59,58 @@ contract OptimisticAsserterLifecycle is Test {
         vm.stopPrank();
     }
 
-    function testBondBelowMinimum() public {
-        vm.expectRevert("Bond amount too low");
-        optimisticAssertor.assertTruthFor(
-            bytes(claimAssertion),
-            address(0),
-            address(0),
-            address(0),
-            defaultCurrency,
-            0,
-            0
-        );
+    function test_AssertionWithDispute() public {
+        vm.startPrank(TestAddress.account1);
+        defaultCurrency.allocateTo(TestAddress.account1, optimisticAssertor.defaultBond());
+        assert(defaultCurrency.balanceOf(TestAddress.account1) >= optimisticAssertor.defaultBond());
+        defaultCurrency.approve(address(optimisticAssertor), optimisticAssertor.defaultBond());
+
+        // Account1 asserts a false claim.
+        bytes32 assertionId =
+            optimisticAssertor.assertTruthFor(
+                bytes(falseClaimAssertion),
+                TestAddress.account1,
+                address(0),
+                address(0),
+                defaultCurrency,
+                optimisticAssertor.defaultBond(),
+                optimisticAssertor.defaultLiveness()
+            );
+        vm.stopPrank();
+
+        // The assertion gets disputed by the disputer, account2.
+        vm.startPrank(TestAddress.account2);
+        defaultCurrency.allocateTo(TestAddress.account2, optimisticAssertor.defaultBond());
+        assert(defaultCurrency.balanceOf(TestAddress.account2) >= optimisticAssertor.defaultBond());
+        defaultCurrency.approve(address(optimisticAssertor), optimisticAssertor.defaultBond());
+
+        optimisticAssertor.disputeAssertionFor(assertionId, TestAddress.account2);
+        vm.stopPrank();
+
+        // In the meantime simulate a vote in the DVM in which the originally disputed price is accepted
+        MockOracleAncillary.QueryPoint[] memory queries = mockOracle.getPendingQueries();
+
+        // There should be only one query.
+        assertEq(queries.length, 1);
+
+        // The query should be for the disputed assertion.
+        assertEq(queries[0].identifier, optimisticAssertor.identifier());
+
+        // Push the resolution price into the mock oracle, a no vote meaning that the assertion is resolved as false.
+        mockOracle.pushPrice(queries[0].identifier, queries[0].time, queries[0].ancillaryData, 0);
+
+        assertEq(optimisticAssertor.settleAndGetAssertion(assertionId), false);
+
+        // The proposer should have lost their bond.
+        assertEq(defaultCurrency.balanceOf(TestAddress.account1), 0);
+
+        // The disputer should have kept their bond and earned half of the proposer's bond.
+        assertEq(defaultCurrency.balanceOf(TestAddress.account2), (optimisticAssertor.defaultBond() * 3) / 2);
```

Comment:
Not sure if I should use the `burnedBondPercentage` instead of this hardcoded 3/2 value

Reply:
better use `burnedBondPercentage` in case we want to adjust it